### ISSUE TITLE
TabSelectedListener incorrect behaviour.

### ISSUE THREAD

**Issue by Kartik1607** [roughike/BottomBar#671]
TabSelectedListener should only fire up when user selects the tab by clicking it. Instead, it seems to fire up even while rotating the screen. 

For example, if I have set default tab to tab 3.
When activity starts, tab 3 is selected. Then, I change tab to tab 2 and load up a fragment.
When the device is rotated, activity and fragment are destroyed. The activity again selects tab 3, which was default and triggers the tab selected listener for tab 3. Then tab 2 is selected and event for that tab is fired up. 

The tab selected listener should only listen to clicks, not other changes like change in device configuration.

**Comment by alez007** [comment]
Yes, happens to me too, my solution for this was to parcel the selected tab right after selection in the listener and then use setDefaultTab method with the selected tab value from the parcelable object.
